File: day2/part1.py
file = open("input.txt", 'r')
lines = file.readlines()
countPwdRespectRule = 0

for l in lines:
    words = l.split(' ')
    pwdLengthRule = words[0]
    pwdMustContainLetter = words[1].replace(':', '')
    pwd = words[2].replace('\n', '')
    # print(pwdLengthRule, pwdLetterRule, pwd)
    pwdLengthRule = pwdLengthRule.split('-')
    pwdMin = int(pwdLengthRule[0])
    pwdMax = int(pwdLengthRule[1])
    letterCountInPwd = 0
    for c in pwd:
        if c == pwdMustContainLetter:
            letterCountInPwd = letterCountInPwd + 1

    if letterCountInPwd <= pwdMax and letterCountInPwd >= pwdMin:
        countPwdRespectRule = countPwdRespectRule + 1

print(countPwdRespectRule)

File: js-solutions/src/day2/part1.js
import { stringInputToArray } from '../utils/string-input-to-array.js';
import { data } from './input1.js';

const dataArray = stringInputToArray(data);
const answer = dataArray.filter((password) => {
    const [lengthReq, letter, pwd] = password.split(' ');
    const [pwdMin, pwdMax] = lengthReq.split('-')
    const letterReq = letter.replace(':', '')

    let letterCountInPwd = 0;
    [...pwd].forEach((c) => {
        if (c === letterReq) {
            letterCountInPwd++;
        }
    })

    return letterCountInPwd >= pwdMin && letterCountInPwd <= pwdMax;
});

console.log(answer.length)
